login-event-publisher/src/main/java/de/intension/events/publishers/rabbitmq/RabbitMqConnectionManager.java

```diff
@@ -0,0 +1,105 @@
+package de.intension.events.publishers.rabbitmq;
+
+import java.io.IOException;
+import java.util.concurrent.TimeoutException;
+
+import org.jboss.logging.Logger;
+import org.keycloak.Config.Scope;
+
+import com.rabbitmq.client.AMQP.BasicProperties;
+import com.rabbitmq.client.BuiltinExchangeType;
+import com.rabbitmq.client.Channel;
+import com.rabbitmq.client.Connection;
+import com.rabbitmq.client.ConnectionFactory;
+
+public enum RabbitMqConnectionManager {
+
+	INSTANCE;
+
+	private static final Logger logger = Logger.getLogger(RabbitMqConnectionManager.class);
+	private static final String EXCHANGE_LOGIN_DETAILS = "login-details";
+
+	public static final BasicProperties PERSISTENT_JSON = new BasicProperties("application/json", null, null, 2, 0,
+			null, null, null, null, null, null, null, null, null);
+
+	private ConnectionFactory connectionFactory;
+	private Connection connection;
+	private Channel channel;
+
+	// Initiates and configures the connection factory using the environment
+	// variables
+	public void init(Scope config, ConnectionFactory connectionFactory) {
+		this.connectionFactory = connectionFactory;
+		this.connectionFactory.setUsername(config.get("rmq-username", "guest"));
+		this.connectionFactory.setPassword(config.get("rmq-password", "guest"));
+		this.connectionFactory.setVirtualHost(config.get("rmq-vhopst", "/"));
+		this.connectionFactory.setHost(config.get("rmq-url", "localhost"));
```

Comment:
I would rename it to rmq-host since it is not an URL in the sense of an internet-address but only the host